### ISSUE TITLE
Add dart fix support for `moved` classes 

### ISSUE THREAD

**Issue by Piinks** [dart-lang/sdk#45393]
We would like to supply fixes that can change a class name based on the method being called.

For example:
```dart
Scaffold.of(context).showSnackBar(SnackBar(content: Text('Snack')));
```

should be migrated to
```dart
ScaffoldMessenger.of(context).showSnackBar(SnackBar(content: Text('Snack')));
```

The is not specific to the *same* method being moved across classes, as we would want to be able to identify and compose fixes that can feasibly migrate `classA.methodA` to `classB.methodB`, or constructors too.

Another example of this would be going from this: 
```dart
BlackListingTextInputFormatter()
```

to this:
```dart
FilteringTextInputFormatter.deny()
```

Not sure if that example should be a different use case, if so we can break it out into a separate issue.

**Comment by craiglabenz** [comment]
I second this feature request, as it would open up a lot of very powerful migrations.

**Comment by bwilkerson** [comment]
I think the following new kind of change would satisfy the requirements expressed here:

---

#### replacedBy

A _replaced by_ change indicates that the specified element was replaced by another element. It has two keys: `kind`, and `newElement`.

The `newElement` key is required. The value of the `newElement` key is an [element]().

---

I think this would allow you to specify that the static method `of` in the class `Scaffold` was replaced by the static method `of` in the class `ScaffoldMessenger`, or that the constructor `BlackListingTextInputFormatter` was replaced by the constructor `FilteringTextInputFormatter.deny`.

Can you think of any other information you might want to have supported by such a change?

**Comment by bwilkerson** [comment]
I have committed https://dart-review.googlesource.com/c/sdk/+/221900, which includes a limited version of a `replaceBy` change. I believe that what's currently implemented is sufficient for both the use cases above and for the SDK use case. I'll update the documentation to include the new change, hopefully today.

I'm going to close this issue in preference of having new issues for any enhancement requests.

**Comment by pq** [comment]
@bwilkerson: maybe cc @natebosch when you update the docs since this ought to get them unblocked?
